### PR TITLE
[Hub Generated] Publish private branch 'azure-kusto/devCM'

### PR DESCRIPTION
**Pull request by matanova8**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/published%2FAzure%2Fazure-rest-api-specs-pr%2Fazure-kusto%2FdevCM...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by matanova8** [comment]
The PR is created based on the updates in the private branch. The updates in the PR has already been reviewed and approved with this PR [Azure/azure-rest-api-specs-pr/10404](https://github.com/Azure/azure-rest-api-specs-pr/pull/10404)

**Comment by matanova8** [comment]
@xiaoxuqi-ms I see that the failure of "Swagger SDK Breaking Change Tracking failed" is due to:
image
BUT 
We didn't change anything about the "SKUDescription.Restrictions" in the last 4 version I checked:
image

How can I make this test to pass?

**Comment by xiaoxuqi-ms** [comment]
@tadelesh , could you help to review this breaking change? Many thanks.

**Comment by xiaoxuqi-ms** [comment]
@matanova8, please make sure the new version has been deployed to prod.

**Comment by xiaoxuqi-ms** [comment]
@lei-msft , could you help to take a look at the API readiness check?

**Comment by raosuhas** [comment]
Can you please update the checklist in the description above ? 

Also this looks like you are adding a new version for an existing service and this PR is not generated properly for a new API version
	
the first commit needs to be the previous api version and the new changes should only be added in the subsequent commits. This is explained in detail here (If using Open API which is recommended)  https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version or here(For manual process) :  https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/83/Manual-Process-Adding-new-API-version . Please remove the "ARMChangesRequested" label once you have fixed this issue.

**Comment by matanova8** [comment]
@raosuhas 
1. I updated the checklist in the description above
2. I read the docs you sent, but didn't understand your request "the first commit needs to be the previous api version and the new changes should only be added in the subsequent commits".
What does it means "commit previous api version"? Can you send an example for a valid PR?+

**Comment by matanova8** [comment]
@leni-msft unfortunatly the hard coded "basic.yaml" wasn't being used:
Hard coded basic.yaml location variable:
image

In practice basic.yaml from last run:
image

**Comment by matanova8** [comment]
FYI - I reverted "[Override basic.yaml](https://github.com/Azure/azure-rest-api-specs/pull/22102/commits/1a32116b596f1e41b7461fbd0977bd75d2ac15fb)" since it included an encoding diff for 2020 examples by mistake. 
Therefore I pushed it again clean: 
[override basic.yaml 2](https://github.com/Azure/azure-rest-api-specs/pull/22102/commits/1cde00c32b00da9569289275b3902bff1d94d03d)

**Comment by xiaoxuqi-ms** [comment]
> Can you please update the checklist in the description above ?
> 
> Also this looks like you are adding a new version for an existing service and this PR is not generated properly for a new API version
> 
> the first commit needs to be the previous api version and the new changes should only be added in the subsequent commits. This is explained in detail here (If using Open API which is recommended) https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version or here(For manual process) : https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/83/Manual-Process-Adding-new-API-version . Please remove the "ARMChangesRequested" label once you have fixed this issue.

Hi @raosuhas , this service enabled API spec pipeline , and this is a release Pull request consisting other prs in dev branches. It may not start with an init commit but actually all the changes for this new API version have been approved. THe API readiness check is the last check to pass. THanks for your review and Ithink  @matanova8 do not need to update the codes as requested.

File: specification/azure-kusto/resource-manager/readme.test.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+### Tag: package-2022-12-29
```
Comment (leni-msft) [inline specification/azure-kusto/resource-manager/readme.test.md:1]:
```suggestion
### Tag: package-2022-12
```

File: specification/azure-kusto/resource-manager/readme.test.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+These settings apply only when `--tag=package-2022-12-29` is specified on the command line.
```
Comment (leni-msft) [inline specification/azure-kusto/resource-manager/readme.test.md:3]:
```suggestion
These settings apply only when `--tag=package-2022-12` is specified on the command line.
```

File: specification/azure-kusto/resource-manager/readme.test.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+``` yaml $(tag) == 'package-2022-12-29'
```
Comment (leni-msft) [inline specification/azure-kusto/resource-manager/readme.test.md:5]:
```suggestion
``` yaml $(tag) == 'package-2022-12'
```

**Comment by MaryGao** [comment]
@harsha The arm-kusto build failed with below [err](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2140833&view=logs&j=f38d9a4f-f3b8-55f9-3f85-04d40f5c1fc2&t=ceba0f41-7307-52b4-7c7a-37e724e7c1f1&l=1929) do you have any idea?

```
06:40:12.548 cmdout 	[automation_generate.sh] ==[ @azure-tools/test-credential ]===============================[ 15 of 16 ]==
06:40:12.548 cmdout 	[automation_generate.sh] "@azure/arm-kusto" is blocked by "@azure-tools/test-credential".
06:40:12.548 cmdout 	[automation_generate.sh] ==[ @azure/core-lro ]============================================[ 16 of 16 ]==
06:40:12.548 cmderr 	[automation_generate.sh] Returned error code: 2
06:40:12.548 cmderr 	[automation_generate.sh] "@azure-tools/test-credential" failed to build.
```

**Comment by xiaoxuqi-ms** [comment]
SDK team is confirming the error with Redmond team to identify the root cause and since this version is reaching to production, I will approve this pr and the API readiness issue will be fixed later.

**Review by xiaoxuqi-ms — APPROVED** [review]

**Pull request merged**